### ISSUE TITLE
Conflict relationship is triggered even when version constrained

### ISSUE THREAD

**Issue by dbent** [KSP-CKAN/CKAN#1245]
This came up in KSP-CKAN/NetKAN#1841 where I attempted to add the following relationship:

```
{
  "conflicts": [
    { "name": "HotRockets", "max_version": "1.0.4" }
  ]
}
```

This still triggered a conflict error even though `HotRockets-1.0.4.1` was installed.

**Comment by pjf** [comment]
Yikes. This happens in the core, and affects both the GUI and Cmdline, yes?

**Comment by dbent** [comment]
Only tested it on the command line, but I'm assuming it's in the relationship resolver so it should affect the GUI as well. I'll double-check.

**Comment by HebaruSan** [comment]
Fixed in #2339.
